server.go

```diff
@@ -198,15 +207,14 @@ func (s *Server) ListenAndServe() error {
 // ListenAndServeTLS listens on the TCP network address s.Addr and then calls
 // Serve to handle requests on incoming TLS connections.
 //
-// If s.Addr is blank, ":smtps" is used.
+// If s.Addr is blank in SMTP mode, ":smtps" is used.
+// If s.Addr is blank in LMTP mode, an error is returned, as there is no default port for lmtps
 func (s *Server) ListenAndServeTLS() error {
-	if s.LMTP {
-		return errTCPAndLMTP
-	}
-
 	addr := s.Addr
-	if addr == "" {
+	if !s.LMTP && addr == "" {
 		addr = ":smtps"
+	} else if s.LMTP && addr == "" {
+		return errNoAddressSpecified
 	}
 
 	l, err := tls.Listen("tcp", addr, s.TLSConfig)
```

Comment:
Shouldn't s.Network handling be duplicated here?

Reply:
Is there a use case for TLS over unix sockets

Reply:
There are other meaningful s.Network values such as tcp4 and tcp6. 
In principle, I believe library should not add any unnecessary restrictions - there is nothing fundamentally wrong in doing TLS-over-Unix sockets and one might even find use-cases for it.